pkg/kubelet/cm/dra/manager_test.go

```diff
@@ -0,0 +1,300 @@
+)
+
+func TestNewManagerImpl(t *testing.T) {
+	kubeClient := fake.NewSimpleClientset()
+	stateFileDirectory := t.TempDir()
+	manager, err := NewManagerImpl(kubeClient, stateFileDirectory)
+	if err != nil {
+		t.Errorf("Failed to create DRA manager: %v", err)
+	}
+
+	if manager.cache == nil {
+		t.Error("Expected cache to be initialized, but it was nil")
+	}
+
+	if manager.kubeClient != kubeClient {
+		t.Error("Expected kubeClient to be set, but it was not")
+	}
+}
+
+func TestGetResources(t *testing.T) {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: "test-namespace",
+		},
+		Spec: v1.PodSpec{
+			ResourceClaims: []v1.PodResourceClaim{
+				{Name: "test-pod-claim-1"},
+				{Name: "test-pod-claim-2"},
+			},
+		},
+	}
+
+	container := &v1.Container{
+		Name: "test-container",
+		Resources: v1.ResourceRequirements{
+			Claims: []v1.ResourceClaim{
+				{Name: "test-pod-claim-1"},
+				{Name: "test-pod-claim-3"},
+			},
+		},
+	}
+
+	kubeClient := fake.NewSimpleClientset()
+	stateFileDirectory := t.TempDir()
+	manager, err := NewManagerImpl(kubeClient, stateFileDirectory)
+	if err != nil {
+		t.Errorf("Failed to create DRA manager: %v", err)
+	}
+
+	_, err = manager.GetResources(pod, container)
+	if err != nil {
+		return
+	}
+}
+
+func TestPrepareResources(t *testing.T) {
+	// Create a fake kubeClient for testing
+	fakeKubeClient := fake.NewSimpleClientset()
+
+	cache, err := newClaimInfoCache(t.TempDir(), "")
+	if err != nil {
+		return
+	}
+
+	// Create a ManagerImpl instance with the fake kubeClient
+	manager := &ManagerImpl{
+		kubeClient: fakeKubeClient,
+		cache:      cache,
+	}
+
+	// Create a Pod with a ResourceClaimSpec
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: "test-namespace",
+			UID:       "test-reserved",
+		},
+		Spec: v1.PodSpec{
+			ResourceClaims: []v1.PodResourceClaim{
+				{Name: "test-pod-claim-1"},
+			},
+		},
+	}
+
+	// Create a ResourceClaim with a DriverName and a ResourceHandle
+	resourceClaim := &resourcev1alpha2.ResourceClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-claim-1",
+			Namespace: "test-namespace",
+		},
+		Spec: resourcev1alpha2.ResourceClaimSpec{
+			ResourceClassName: "test-class",
+		},
+		Status: resourcev1alpha2.ResourceClaimStatus{
+			DriverName: "test-driver",
+			Allocation: &resourcev1alpha2.AllocationResult{
+				ResourceHandles: []resourcev1alpha2.ResourceHandle{
+					{Data: "test-data"},
+				},
+			},
+			ReservedFor: []resourcev1alpha2.ResourceClaimConsumerReference{
+				{UID: "test-reserved"},
+			},
+		},
+	}
+
+	_, err = fakeKubeClient.ResourceV1alpha2().ResourceClaims(pod.Namespace).Create(context.Background(), resourceClaim, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("failed to create ResourceClaim: %+v", err)
+	}
+
+	// Call the PrepareResources method
+	err = manager.PrepareResources(pod)
+	if err != nil {
+		t.Fatalf("PrepareResources failed: %+v", err)
+	}
```

Comment:
This looks strange. How `PrepareResources` can possibly succeed if DRA Kubelet plugin is not running? There must be a bug somewhere in the code.

Reply:
#118468 this issues is created to track this place? or it's different problem

Reply:
I think it's a different problem. TestPrepareResources doesn't test anything. I think it silently returns at this place:
```	cache, err := newClaimInfoCache(t.TempDir(), "")
	if err != nil {
		return
	}
```
If you replace `return` with `t.Fatalf` call, you'll see it.

Reply:
You can also look at the test coverage to see how manager.go code is covered by the tests. In my setup coverage viewer shows 13.3% coverage for manager.go, which means most of the code is not tested.

Reply:
will check this later today. can you review the rest of the code and point out where do I need to change/address as well? would solve it when I back to home. Thanks!

Reply:
Overall observation after briefly looking at the code - you should avoid bare `return` on errors. Use either `t.Fatalf` or `t.Errorf` depending on severity of the failure.  `return` just hides the error.